### PR TITLE
Add vw_types.natvis to CMake definition in Visual Studio

### LINKED ISSUE #2118: vw_types.natvis should be added to CMake definition when generator is visual studio
In order to make use of the natvis type definitions, they need to be part of the generated visual studio project. They should be added only when the generator is visual studio.

See here for an example: https://github.com/microsoft/GSL/blob/master/CMakeLists.txt

### PR DESCRIPTION
**Pull request by Bharat123rox**
Fixes #2118 by adding `vw_types.natvis` when the generator is Visual Studio

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -198,6 +198,22 @@ add_subdirectory(cluster)
...
+if (CMAKE_VERSION VERSION_GREATER 3.7.8)
+    if (MSVC_IDE)
+        option(VS_ADD_NATIVE_VISUALIZERS "Configure project to use Visual Studio native visualizers" TRUE)
+    else()
+        set(VS_ADD_NATIVE_VISUALIZERS FALSE CACHE INTERNAL "Native visualizers are Visual Studio extension" FORCE)
+    endif()
+
+    # Add natvis file to the library so it will automatically be loaded into Visual Studio
+    if(VS_ADD_NATIVE_VISUALIZERS)
+        target_sources(vowpal_wabbit INTERFACE
```
Comment (jackgerrits) [inline CMakeLists.txt]: This is being added as a source to the project itself, what behavior does that implement? In VS, does it show up as a source file on the solution?

File: CMakeLists.txt [hunk 2]
```diff
@@ -198,6 +198,22 @@ add_subdirectory(cluster)
 add_subdirectory(library)
 add_subdirectory(vowpalwabbit)
 
+# Add natvis file if the Generator is Visual Studio
+if (CMAKE_VERSION VERSION_GREATER 3.7.8)
+    if (MSVC_IDE)
+        option(VS_ADD_NATIVE_VISUALIZERS "Configure project to use Visual Studio native visualizers" TRUE)
```
Comment (jackgerrits) [inline CMakeLists.txt]: What is the reasoning behind making this an option? Is there a time when you would not want to do this, or to force it to be on?

**Comment by jackgerrits** [comment]
@Bharat123rox have you had a chance to look at my comments?

**Comment by Bharat123rox** [comment]
@jackgerrits yes, but I'm still trying to make sense of what a CMakeList does :stuck_out_tongue:

**Comment by jackgerrits** [comment]
Not a problem, thanks for opening this. Let me know if you want helping working through this change

**Comment by jackgerrits** [comment]
@Bharat123rox, would you like me to have a crack at making the changes?

**Comment by Bharat123rox** [comment]
Sorry @jackgerrits for the delayed reply, please take a look at this yourself as I cannot devote time to this as I had originally planned to, also please tell me if I should keep this PR open or not.

**Comment by jackgerrits** [comment]
Not a problem. Thanks for starting on this! I will branch off your PR and open a new one when I have it ready

**Comment by jackgerrits** [comment]
Hi @Bharat123rox, I took your branch and made the edits and have opened a new PR #2186. I'm going to close this PR, but I appreciate you getting started on this issue. It's great having you involved in the community!

**Pull request closed without merging**